microsoft-edge/webview2/how-to/winrt-js-conversion.md

```diff
@@ -0,0 +1,139 @@
+| `IVector<T>`, `IVectorView<T>`, `IObservableVector<T>` | `Array` and JavaScript object | If a `RuntimeClass` instance implements `vector` interfaces, then it is represented in JavaScript as the usual object described below, but also will act like a JavaScript array.  Reads and writes are performed live on the underlying WinRT vector object.
+| `IMap<K,V>`, `IMapView<K,V>`, `IObservableMap<K,V>` | JavaScript object | If a `RuntimeClass` instance implements `map` interfaces, then it is represented in JavaScript as the usual object described below, but also has properties with name and values from the underlying WinRT map object.  Reads and writes are performed live on the underlying WinRT map object. |
+| `Enum` | JavaScript object | An enum type is represented as a JavaScript object.  Each enum value is a `Number` property on the JavaScript object. |
+| `Struct` | JavaScript object | A `Struct` type is converted to a JavaScript object that has property names which correspond to the `Struct` type member names.  This is a two-way conversion. |
+| `Namespace` | JavaScript object | A namespace is represented as a JavaScript object that has a property for each child namespace, enum type, and `RuntimeClass`. |
+| `Class` | JavaScript object | A `RuntimeClass` class is converted to a JavaScript object that has the same methods, properties, and events. |
+| `Interface` | JavaScript object | A `RuntimeClass` interface is converted to a JavaScript object that has the same methods, properties, and events.  There is no support for implementing an interface in JavaScript. |
+| Class static member | JavaScript object property | See below. |
+| Class constructor | JavaScript constructor and function | See below. |
+
+See also:
+* [RuntimeClass class](https://learn.microsoft.com/en-us/cpp/cppcx/wrl/runtimeclass-class)
+
+
+<!-- ------------------------------ -->
+#### Class static members
+
+<!--
+original wording:
+Runtimeclasses with static methods, properties, or events, are represented as properties of their namespace with the static methods, properties, and events as properties on that JavaScript object.
+
+that might mean:
+
+The runtime class that has static properties, methods, or events is a property of the namespace.  Each static property, method, or event is a property of the JavaScript Object.
+
+A `RuntimeClass` class that has static properties, methods, or events is represented as a property of the corresponding JavaScript namespace object.  The static methods, properties, and events are represented as properties on the JavaScript namespace object.
+-->
+A runtime class that has static properties, static methods, or static events is represented as a property of the namespace.  Each static property, method, or event is represented as a property on that JavaScript object.
```

Comment:
Sorry for my confusing wording. Here are some potential updates to the wording and then trying to explain what I mean more below

```suggestion
A runtime class that has static properties, static methods, or static events is represented as a property of the namespace.  Each static property, static method, and static event is represented as a property on that JavaScript object of the runtimeclass.
```

Eg for the WinRT API static method `Windows.Foundation.Uri.EscapeComponent`, `Windows.Foundation` is the namespace, `Uri` is the runtimeclass, and `EscapeComponent` is the static method. In JavaScript it looks similar: `chrome.webview.hostObjects.Windows.Foundation.Uri.EscapeComponent` where the EscapeComponent is a JS method that is a property on the JS object for the Uri runtimeclass, which is a property on the JS object for the Foundation namespace.

Reply:
snapshot of result i'm trying:
```
A runtime class that has static properties, static methods, or static events is represented as a property of the namespace. Each static property, static method, and static event is represented as a property on that JavaScript object of the runtimeclass.

For example, for the WinRT API static method `Windows.Foundation.Uri.EscapeComponent`:
*  `Windows.Foundation` is the namespace.
*  `Uri` is the runtimeclass.
*  `EscapeComponent` is the static method.

In JavaScript, the representation looks similar: `chrome.webview.hostObjects.Windows.Foundation.Uri.EscapeComponent`:
*  `EscapeComponent` is a JavaScript method that's a property on the JavaScript object for the `Uri` runtimeclass.
*  The `Uri` runtimeclass is a property on the JavaScript object for the `Foundation` namespace.
```